packages/front-end/pages/saved-groups.tsx

```diff
@@ -1,23 +1,61 @@
 import { SavedGroupInterface } from "back-end/types/saved-group";
 import React, { useState } from "react";
-import { FaPencilAlt } from "react-icons/fa";
 import Button from "../components/Button";
 import SavedGroupForm from "../components/SavedGroupForm";
 import { GBAddCircle } from "../components/Icons";
 import LoadingOverlay from "../components/LoadingOverlay";
 import { ago } from "../services/dates";
 import { useDefinitions } from "../services/DefinitionsContext";
 import usePermissions from "../hooks/usePermissions";
+import MoreMenu from "../components/Dropdown/MoreMenu";
+import DeleteButton from "../components/DeleteButton/DeleteButton";
+import { useFeaturesList } from "../services/features";
+import Link from "next/link";
+import { useAuth } from "../services/auth";
+
+const getSavedGroupMessage = (featuresUsingSavedGroups: string[]) => {
+  return async () => {
```

Comment:
Does this need to be async?

Reply:
Kinda. It doesn't need to be async in the sense that we're not actually doing anything inside of this that requires us to await something.

However, within `DeleteButton.tsx`, that is built on the assumption that this message is going to return a promise.

I deviated from the existing pattern (found in `[mid].tsx` and `pages/segments/index.tsx`. In those components, we're fetching a list of metrics/segments. For features, it felt silly to fetch all of the features (potentially multiple times) since we have the `useFeatureList` hook.

The second benefit of deviating from the existing pattern is in this component. we're introducing a new prop `canDelete` that is passed in to `DeleteButton`. This is used to disable the submit button since in this case, if a saved group is being used by a feature, we're not allowing the user to delete the saved group.

If we were to fetch the list of features via the `getConfirmationContent` function, we'd have to fetch the list of features again in order to return a value for `canDelete`. 

Of course, if you have a suggestion on a better way to handle this, I'm all ears. It definitely feels like an anti-pattern to just wrap something in an async function that doesn't await something.